File: apps/cargo-tracker/src/CargoTracker.Web/Shared/Domain/Model/ShipperId.cs
namespace CargoTracker.Shared.Domain.Model;

/// <summary>荷主識別子（共有カーネル）。Guid ベースで Booking / Shipper コンテキストが共有する。</summary>
public sealed record ShipperId(Guid Value)
{
    public static ShipperId New() => new(Guid.NewGuid());
}

File: apps/cargo-tracker/backend/bookingms/src/main/java/com/example/cargotracker/booking/domain/model/valueobjects/ShipperId.java
package com.example.cargotracker.booking.domain.model.valueobjects;

/** 荷主の識別子（Booking Context 側の型）。 */
public record ShipperId(String value) {

    public ShipperId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("荷主 ID は必須です");
        }
    }
}
